README.md

```diff
@@ -1,48 +1,162 @@
+## Training and evaluating an FDT agent
+The script `train_agent_babyai.py` contains everything needed to 
+- generate a training dataset
+- train an FDT agent
+- run periodic evaluation during training and save results
 
-### Example
-```sh
-python src/_datasets.py --env_name "BabyAI-GoToRedBallGrey-v0" --num_episodes 1000 --include_timeout False
+To run it with default arguments, simply execute 
+```bash
+python scripts/train_agent_babyai.py
 ```
 
-## Generating language feedback for an existing BabyAI dataset
+The script also supports a number of arguments. The main ones are highlighted below; for a complete list refer to `src/utils/argparsing.py`.
+
+### Specifying a BabyAI level
+The `--level` argument determines which BabyAI level to use, defaulting to "GoToRedBallGrey". A list of levels can be found on [this page](https://minigrid.farama.org/environments/babyai/) under "Registered Configurations". 
+
+### Controlling dataset creation
+To avoid creating a new dataset everytime the script is run, you can use the `--load_existing_dataset` argument - this will check if a dataset matching all the relevant arguments exists, and load it if so. By default, datasets are generated using a random policy - to instead sample data from the training process of a PPO agent, set `--policy` to "ppo". For both random and PPO policies, `--num_steps` controls the number of environment steps to include in the training dataset. Finally, to exclude episodes that end as a result of timing out, set `--include_timeout` to false. 
+
+### Controlling what action prediction is conditioned on
+The original decision transformer uses return-to-go (RTG) to condition action predictions during training. Our architecture allows conditioning on RTG, mission string, language feedback, or any combination - you can control this by setting the `--use_rtg`, `--use_mission` and `--use_feedback` arguments. You can also control the provision of feedback and mission strings at *inference* time using the `--feedback_at_inference` and `--mission_at_inference` arguments.
 
-### Specifying the dataset
-Rather than providing the name of the dataset, simply provide the same argument values as when you created the dataset (or in case the dataset doesn't exist yet, it will be created now), e.g. ``--env-name "BabyAI-GoToRedBallGrey-v0" --num_episodes 1000 --include_timeout False --seed 42``.
+### Controlling feedback generation
+If conditioning on feedback (see previous section), you can determine which *type* of feedback is used by setting the `--feedback_mode` argument to one of 'all', 'rule', 'task', 'numerical' or 'random'. For an explanation of what these feedback types mean, see section 3 and appendix A.1 of the paper. If using random feedback, you can additionally specify the type of random feedback through `--random_mode`, which can be either 'english' (uses random words from the BabyAI vocabulary) or 'lorem' (uses lorem ipsum text).
```

Comment:
1. Maybe we could briefly explain what numerical does?
2. 'english' uses random (nonsensical) English sentences that _don't_ contain any of the words in the BabyAI vocabulary

Reply:
oops my bad 😅